src/NHibernate.Test/Ado/GenericBatchingBatcherFixture.cs

```diff
@@ -0,0 +1,155 @@
+﻿using System;
+using System.Collections;
+using System.Linq;
+using NHibernate.AdoNet;
+using NHibernate.Cfg;
+using NHibernate.Dialect;
+using NUnit.Framework;
+using Environment = NHibernate.Cfg.Environment;
+
+namespace NHibernate.Test.Ado
+{
+	[TestFixture]
+	public class GenericBatchingBatcherFixture : TestCase
+	{
+		protected override string MappingsAssembly => "NHibernate.Test";
+
+		protected override IList Mappings => new[] {"Ado.VerySimple.hbm.xml"};
+
+		protected override void Configure(Configuration configuration)
+		{
+			configuration.SetProperty(Environment.BatchStrategy, typeof(GenericBatchingBatcherFactory).AssemblyQualifiedName);
+			configuration.SetProperty(Environment.GenerateStatistics, "true");
+			configuration.SetProperty(Environment.BatchSize, "1000");
+		}
+
+		protected override bool AppliesTo(Dialect.Dialect dialect)
+		{
+			return !(dialect is FirebirdDialect) &&
+			       !(dialect is Oracle8iDialect) &&
+			       !(dialect is MsSqlCeDialect);
+		}
+
+		[Test]
+		public void MassiveInsertUpdateDeleteTest()
+		{
+			var totalRecords = 1000;
+			BatchInsert(totalRecords);
+			BatchUpdate(totalRecords);
+			BatchDelete(totalRecords);
+
+			DbShoudBeEmpty();
+		}
+
+		[Test]
+		public void BatchSizeTest()
+		{
+			using (var sqlLog = new SqlLogSpy())
+			using (var s = Sfi.OpenSession())
+			using (var tx = s.BeginTransaction())
+			{
+				s.SetBatchSize(5);
+				for (var i = 0; i < 20; i++)
+				{
+					s.Save(new VerySimple { Id = 1 + i, Name = $"Fabio{i}", Weight = 1.45 + i });
+				}
+				tx.Commit();
+
+				var log = sqlLog.GetWholeLog();
+				Assert.That(4, Is.EqualTo(FindAllOccurrences(log, "Batch commands:")));
+			}
+			Cleanup();
+		}
+
+		private void BatchInsert(int totalRecords)
+		{
+			Sfi.Statistics.Clear();
+			using (var s = Sfi.OpenSession())
+			using (var tx = s.BeginTransaction())
+			{
+				for (var i = 0; i < totalRecords; i++)
+				{
+					s.Save(new VerySimple {Id = 1 + i, Name = $"Fabio{i}", Weight = 1.45 + i});
+				}
+				tx.Commit();
+			}
+			Assert.That(Sfi.Statistics.PrepareStatementCount, Is.EqualTo(1));
```

Comment:
This does not seem to check anything. Changing the batch size to 1 in `Configure` still allow the test to succeed this assert. That seems to be an old false assumption that this property would allow to get the number of command executed.

Reply:
I noticed that. The problem is that the only way to have this corrected (at least what I found) is to use `SqlLogSpy` as in the `BatchSizeTest` method, which I didn't use as it would flood the logs. Is there any other alternative?

Reply:
Unfortunately I do not know another way in current NHibernate state. And yes we also have a test log trouble, this log already being to much flooded by the tests, causing CI platforms log checking to be a bit painful. (Well, TeamCity is better than others on this subject.)